### ISSUE TITLE
[Request] Dimension Specific difficulty multipliers

### ISSUE THREAD

**Issue by Conga0** [SilentChaos512/ScalingHealth#154]
Currently you can increase the rate at which difficulty increases but you can't give a direct difficulty increase for being in a dimension; for example being inside the nether "doubles" your current difficulty to all mobs.

I'd like this feature greatly for a modpack i'm creating

Example:
(make difficulty be twice as high when in the nether)
-1 2

(make difficulty be only 75% of its normal amount when inside the end
1 0.75

tl;dr 
Please add difficulty multipliers or flat amounts when a player is in a specific dimension

**Comment by SonicX8000** [comment]
Would a choice of adding a flat number to add on top of the difficulty be an option instead of a multiplier? Like...

Overworld => no extra difficulty

Nether => +20 Extra Difficulty

The End => +40 Extra Difficulty

etc.

The HUD could be something like this... note that this is just an example image and this is only with adding the value on top of your current difficulty.
![](https://cdn.discordapp.com/attachments/377171551155650560/544239921368662026/difficultyaddupdim.PNG)

Because this is extra difficulty being added... this bypasses the difficulty cap, so if you have it set at 500 and you being in a dimension adds an extra +25 difficulty, your total difficulty will be 525 as long as you remain in that dimension. Leaving and going back to the Overworld will remove the +25 and will only show you having 500 difficulty.

If using a multiplier... it would say '500 x1.1' which total would be 550, I think if in the Nether. '500 x1.2' which would be 600 if in The End. Adding to it via a flat number just feels like you have more control with it as a multiplier would be quite an increase as your actual difficulty goes up.

**Comment by SilentChaos512** [comment]
Added in version 1.3.35. You can add, subtract, multiply, or divide by a given value. It's dimension ID, then the operation and value together (so, only one space) Examples:

```
1 +20
0 -10
-1 *1.5
0 /2
```

I chose not to add anything to the difficulty meter for this, since there are multiple factors that affect the value. But I did uncap the text in b40abd144d4036f4d71dcf831fe5afbefa396dba, so it will display the actual area difficulty, even when it's beyond the "max value". The bar still respects the max value of course.
